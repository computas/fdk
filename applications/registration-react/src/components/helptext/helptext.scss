@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-reg-helptext {
  background-color: $color6;
  border-radius: 5px;
  font-size: 1.4rem;
  strong {
    font-size: 1.8rem;
  }
  .fdk-reg-helptext-more {
    text-decoration: underline;
    white-space: nowrap;
  }
}

.fdk-obligatorisk {
  font-size: 1.1rem;
  color: #ffffff;
  background-color: #4B555F;
  padding: .1em .6em;
  border-radius: 2px;
  position: relative;
  top: -2px;
}