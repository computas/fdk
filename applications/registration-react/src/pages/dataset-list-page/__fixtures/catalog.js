export default {
  item: {
    id: '910244132',
    uri: 'http://brreg.no/catalogs/910244132',
    title: {
      nb: 'Datakatalog for RAMSUND OG ROGNAN REVISJON'
    },
    description: {},
    publisher: {
      uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
      name: 'RAMSUND OG ROGNAN REVISJON'
    }
  }
};
