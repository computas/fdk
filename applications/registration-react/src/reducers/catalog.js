import {
  CATALOG_REQUEST,
  CATALOG_SUCCESS,
  CATALOG_FAILURE
} from '../constants/ActionTypes';

export default function catalog(
  state = { isFetchingCatalog: false, catalogItem: null },
  action
) {
  switch (action.type) {
    case CATALOG_REQUEST:
      return {
        ...state,
        isFetchingCatalog: true
      };
    case CATALOG_SUCCESS:
      return {
        ...state,
        isFetchingCatalog: false,
        catalogItem: action.response.data
      };
    case CATALOG_FAILURE:
      return {
        ...state,
        isFetchingCatalog: false,
        catalogItem: null
      };
    default:
      return state;
  }
}
