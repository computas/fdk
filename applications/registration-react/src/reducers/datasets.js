import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../constants/ActionTypes';

export default function datasets(
  state = { isFetchingDatasets: false, datasetItems: null },
  action
) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        isFetchingDatasets: true
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: action.response.data
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: null
      };
    default:
      return state;
  }
}
