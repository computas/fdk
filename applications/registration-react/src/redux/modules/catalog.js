import _ from 'lodash';
import { fetchActions } from '../fetchActions';

const CATALOG_REQUEST = 'CATALOG_REQUEST';
const CATALOG_SUCCESS = 'CATALOG_SUCCESS';
const CATALOG_FAILURE = 'CATALOG_FAILURE';

function shouldFetch(metaState) {
  const threshold = 60 * 1000; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      (metaState.lastFetch || 0) < Date.now() - threshold)
  );
}

export function fetchCatalogIfNeeded(catalogId) {
  return (dispatch, getState) => {
    if (shouldFetch(_.get(getState(), ['catalog', 'meta', catalogId]))) {
      dispatch(
        fetchActions(`/catalogs/${catalogId}`, [
          { type: CATALOG_REQUEST, meta: { catalogId } },
          { type: CATALOG_SUCCESS, meta: { catalogId } },
          { type: CATALOG_FAILURE, meta: { catalogId } }
        ])
      );
    }
  };
}

const initialState = {};

export default function catalog(state = initialState, action) {
  switch (action.type) {
    case CATALOG_REQUEST:
      return {
        ...state,
        meta: {
          [action.meta.catalogId]: {
            isFetching: true,
            lastFetch: null
          }
        }
      };
    case CATALOG_SUCCESS:
      return {
        ...state,
        items: {
          [action.meta.catalogId]: action.payload
        },
        meta: {
          [action.meta.catalogId]: {
            isFetching: true,
            lastFetch: null
          }
        }
      };
    case CATALOG_FAILURE:
      return {
        ...state,
        meta: {
          [action.meta.catalogId]: {
            isFetching: true,
            lastFetch: null
          }
        }
      };
    default:
      return state;
  }
}
