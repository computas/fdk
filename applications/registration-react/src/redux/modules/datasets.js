import _ from 'lodash';
import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../../constants/ActionTypes';
import { fetchActions } from '../fetchActions';

function shouldFetch(metaState) {
  const threshold = 60 * 1000; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      (metaState.lastFetch || 0) < Date.now() - threshold)
  );
}

export function fetchDatasetsIfNeeded(catalogId) {
  return (dispatch, getState) =>
    shouldFetch(_.get(getState(), ['datasets', catalogId, 'meta'])) &&
    dispatch(
      fetchActions(`/catalogs/${catalogId}/datasets?size=100`, [
        { type: DATASETS_REQUEST, meta: { catalogId } },
        { type: DATASETS_SUCCESS, meta: { catalogId } },
        { type: DATASETS_FAILURE, meta: { catalogId } }
      ])
    );
}

const initialState = {};

export default function datasets(state = initialState, action) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        [action.meta.catalogId]: {
          meta: {
            isFetching: true,
            lastFetch: null
          }
        }
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        [action.meta.catalogId]: {
          items: _.get(action.payload, ['_embedded', 'datasets']),
          meta: {
            isFetching: false,
            lastFetch: Date.now()
          }
        }
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        [action.meta.catalogId]: {
          meta: {
            isFetching: false,
            lastFetch: null
          }
        }
      };
    default:
      return state;
  }
}

export const getDatasetItemsByCatalogId = (datasets, catalogId) =>
  _.get(datasets, [catalogId, 'items']);
