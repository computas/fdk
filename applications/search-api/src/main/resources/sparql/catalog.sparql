PREFIX dcat: <http://www.w3.org/ns/dcat#>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX dc: <http://purl.org/dc/elements/1.1/>
PREFIX http: <http://www.w3.org/2011/http#>
PREFIX fo: <http://www.w3.org/1999/XSL/Format#>
PREFIX foaf: <http://xmlns.com/foaf/0.1/>
prefix owl: <http://www.w3.org/2002/07/owl#>
prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
prefix vcard: <http://www.w3.org/2006/vcard/ns#>
prefix er: <http://data.brreg.no/meta/>
prefix xsd: <http://www.w3.org/2001/XMLSchema#>
prefix schema: <http://schema.org/>

DESCRIBE ?publisher ?dataset ?distribution ?catalog ?contact ?temporal ?spatial
WHERE {
  ?catalog a dcat:Catalog;
  dct:publisher ?publisher;
  dcat:dataset ?dataset.

  OPTIONAL {?dataset dcat:distribution ?distribution}
  OPTIONAL {?dataset dcat:contactPoint ?contact}
  OPTIONAL {?dataset dct:temporal ?temporal}
  OPTIONAL {?dataset dct:spatial ?spatial}

  FILTER (?catalog = <%s>)

}
