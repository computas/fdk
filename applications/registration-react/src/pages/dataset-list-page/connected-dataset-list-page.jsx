import { connect } from 'react-redux';
import { fetchHelptextsIfNeeded } from '../../redux/modules/helptexts';
import { fetchDatasetsIfNeeded } from '../../redux/modules/datasets';
import { fetchCatalogIfNeeded } from '../../redux/modules/catalog';
import { DatasetsListPage } from './dataset-list-page';

const mapStateToProps = ({ helptexts, catalog, datasets }) => {
  const { helptextItems } = helptexts || {};
  return {
    helptextItems,
    catalog,
    datasets
  };
};

const mapDispatchToProps = dispatch => ({
  fetchHelptextsIfNeeded: () => dispatch(fetchHelptextsIfNeeded()),
  fetchCatalogIfNeeded: catalogId => dispatch(fetchCatalogIfNeeded(catalogId)),
  fetchDatasetsIfNeeded: catalogId => dispatch(fetchDatasetsIfNeeded(catalogId))
});

export const ConnectedDatasetsListPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(DatasetsListPage);
