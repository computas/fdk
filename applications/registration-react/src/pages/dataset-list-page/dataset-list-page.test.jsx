import React from 'react';
import { shallow } from 'enzyme';
import { DatasetsListPage } from './dataset-list-page';
import datasets from './__fixtures/datasets';
import helptext from '../../../test/fixtures/helptext';
import catalog from './__fixtures/catalog';

let defaultProps;
let wrapper;
let dispatch;

beforeEach(() => {
  const fetchHelptextsIfNeeded = jest.fn();
  const fetchCatalogIfNeeded = jest.fn();
  const fetchDatasetsIfNeeded = jest.fn();
  defaultProps = {
    dispatch,
    datasets,
    helptextItems: helptext.helptextItems,
    catalog,
    fetchHelptextsIfNeeded,
    fetchCatalogIfNeeded,
    fetchDatasetsIfNeeded
  };
  wrapper = shallow(<DatasetsListPage {...defaultProps} />);
});

test('should render DatasetsListPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
