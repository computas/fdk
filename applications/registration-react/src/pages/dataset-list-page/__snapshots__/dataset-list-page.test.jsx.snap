// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DatasetsListPage correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    />
  </div>
  <div
    className="row mb-2 mb-md-5"
  />
</div>
`;
